flytekit/types/file/file.py

```diff
@@ -217,46 +248,78 @@ def to_literal(
 
         if python_val is None:
             raise AssertionError("None value cannot be converted to a file.")
+
+        # information used by all cases
+        meta = BlobMetadata(type=self._blob_type(format=FlyteFilePathTransformer.get_format(python_type)))
+
         if isinstance(python_val, FlyteFile):
-            # If the object has a remote source, then we just convert it back.
+            source_path = python_val.path
+
+            # If the object has a remote source, then we just convert it back. This means that if someone is just
+            # going back and forth between a FlyteFile Python value and a Blob Flyte IDL value, we don't do anything.
             if python_val._remote_source is not None:
-                meta = BlobMetadata(type=self._blob_type(format=self.get_format(python_type)))
                 return Literal(scalar=Scalar(blob=Blob(metadata=meta, uri=python_val._remote_source)))
 
-            source_path = python_val.path
-            if python_val.remote_path is False:
-                # If the user specified the remote_path to be False, that means no matter what, do not upload
+            # If the user specified the remote_path to be False, that means no matter what, do not upload. Also if the
+            # path given is already a remote path, say https://www.google.com, the concept of uploading to the Flyte
+            # blob store doesn't make sense.
+            if python_val.remote_path is False or ctx.file_access.is_remote(source_path):
```

Comment:
should we do the is_remote check? Why not just upload?

Reply:
But if a task
```
def t1() -> FlyteFile:
  return FlyteFile("https://www.google.com")
```
what does it mean to upload that?